src/components/API.js

```diff
@@ -0,0 +1,110 @@
+export class Api {
+  constructor(options) {
+    // constructor body
+  }
+
+  _checkResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.reject(`Error: ${res.status}`);
+  }
+
+  getInitialCards() {
+    return fetch("https://around-api.en.tripleten-services.com/v1/cards", {
+      headers: {
+        authorization: "14d5451e-ea68-4211-8036-23527b03c3ac",
+      },
+    }).then(this._checkResponse);
+  }
+  getUserInfo() {
+    return fetch("https://around-api.en.tripleten-services.com/v1/users/me", {
+      headers: {
+        authorization: "14d5451e-ea68-4211-8036-23527b03c3ac",
+      },
+    }).then(this._checkResponse);
```

Comment:
**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.